pkg/proc/variables.go

```diff
@@ -311,7 +311,7 @@ func GoroutinesInfo(dbp *Target, start, count int) ([]*G, int, error) {
 	threads := dbp.ThreadList()
 	for _, th := range threads {
 		if th.Blocked() {
-			continue
+			//continue
```

Comment:
This is on purpose. We either aren't sure that we can read memory from thread if Blocked is true or we don't want to because it could be wrong (runtime.clone). We can't do this.

Reply:
Okey, I  should fix the `th.Blocked` actually.

Reply:
> This is on purpose. We either aren't sure that we can read memory from thread if Blocked is true or we don't want to because it could be wrong (runtime.clone). We can't do this.

1. There is an state that the `th.Blocked()` returned true, and i checked than the fn.Name is "runtime.futex”. **But** the `g, _ := GetG(th)` still can get the right `g` and the `proc.G{Status: 0x02}` in `g` means it is running.
2. I checked `GetG()` and it already has the code to check whether it's in ` runtime.clone`: https://github.com/go-delve/delve/blob/788c41be761b9c4b84979f9e8b0d9416dd7bcfb3/pkg/proc/variables.go#L242

So, why there need to check the threads is Blocked? We can use `GetG` to check whether there is a `g` in thread directly.

Reply:
I'm not sure why any of this matters.

Reply:
When the thread going to `runtime·futex`, there is no _p_ maybe, but it still have _g_ and _g_'s status is still running. I think it is save to get the _g_ maybe, otherwise we may lost some _g_ when listing gorutines at that instant.